src/core/qgsaggregatecalculator.cpp

```diff
@@ -77,22 +80,20 @@ QVariant QgsAggregateCalculator::calculate( QgsAggregateCalculator::Aggregate ag
   else
     lst = expression->referencedColumns();
 
-  QgsFeatureRequest request = QgsFeatureRequest()
-                              .setFlags( ( expression && expression->needsGeometry() ) ?
-                                         QgsFeatureRequest::NoFlags :
-                                         QgsFeatureRequest::NoGeometry )
-                              .setSubsetOfAttributes( lst, mLayer->fields() );
+  requestCopy.setFlags( ( expression && expression->needsGeometry() ) ?
+                        QgsFeatureRequest::NoFlags :
+                        QgsFeatureRequest::NoGeometry )
+  .setSubsetOfAttributes( lst, mLayer->fields() );
   if ( !mFilterExpression.isEmpty() )
-    request.setFilterExpression( mFilterExpression );
+    requestCopy.setFilterExpression( mFilterExpression );
```

Comment:
@m-kuhn What's your thoughts on this? The danger here is that if someone specifies a feature request with an existing filter expression or filter fids etc, it'll get silently overridden by this.

I'm 50/50 on it. I don't think it's a big issue so long as the dox for the the request parameter and for setFilter explicitly state this situation using "\warn" tags

Reply:
Yeah I didn't want to do this but Mr. Kuhn was proving insistent with this instead of the feature Ids. But as stated in the emails if someone sets and expression it overrides the filter lists. This is why I wanted to address this issue.

Initially using the filter list was simple but since it was lower than the expression in  the order this meant that the expression was overridden but the feature list.

Once again in both cases we are at a loss. The tests that I did with my previous proposal worked somehwat fine but could lead to segfault if stacking was allowed and no expression was provided.

I don't mind using either to be honest.

Reply:
Oh, I wasn't aware there already is a filter possibility (via a mFilterExpression member) on the aggregate calculator.

In this case, shouldn't the feature ids just be inserted in there?
Or the list of filter ids be injected via setParameters too?

Reply:
> Oh, I wasn't aware there already is a filter possibility (via a mFilterExpression member) on the aggregate calculator.
> 
> In this case, shouldn't the feature ids just be inserted in there?
> Or the list of filter ids be injected via setParameters too?

...so I'll just revert everything to its original state. :confused:

Reply:
Ignore comment about reseting,

As far as the expression part yes this could be done but as I stated previously currently the expressions and variables are more QVariant and support for QVariantList is limited. This implies recasting everything for storage and to get it back as a QgsFeatureIds, this also mean that I would need to find a way to retrieve that list as a variable and currently support for list retrieval is not implemented from what I tested, but I may be wrong.

This is why I commented out the use of Symbol_features (can't recall the name) in my other PR.

Reply:
@m-kuhn So should I revert to filter ids or leave it as a custom request?